app/renderer/actions/Session.js

```diff
@@ -724,6 +727,54 @@ export function setSavedServerParams () {
   };
 }
 
+export function setStateFromAppiumFile (newFilepath = null) {
+  return (dispatch) => {
+    // no "fs" means not a browser. do nothing
+    if (!fs) {
+      return;
+    }
+    try {
+      let filePath = newFilepath;
+      if (!newFilepath) {
+        const lastArg = process.argv[process.argv.length - 1];
+        if (!lastArg.startsWith('filename=')) {
+          return;
+        }
+        filePath = lastArg.split('=')[1];
+      }
+      const filePathStorageKey = 'last_opened_file';
+      if (sessionStorage.getItem(filePathStorageKey) === filePath) {
+        // file was opened already, do nothing
+        return;
+      }
+      const appiumJson = JSON.parse(fs.readFileSync(filePath, 'utf8'));
```

Comment:
how is it possible to call `fs` from the renderer?

Reply:
Magic!